### PR TITLE
RDBS-79 Modify HttpClientHandler connection limit to not rely on the default

### PR DESCRIPTION
**Pull request by haludi**
https://issues.hibernatingrhinos.com/issue/RDBS-79

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -38,6 +38,8 @@ namespace Raven.Client.Http
 {
     public class RequestExecutor : IDisposable
     {
+        private const int MinConnectionLimit = 1024 * 10;
```
Comment (ppekrol) [inline src/Raven.Client/Http/RequestExecutor.cs]: rename to `DefaultConnectionLimit`

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 2]
```diff
@@ -1780,17 +1782,15 @@ public virtual void Dispose()
...
-            var httpMessageHandler = new HttpClientHandler();
+            var httpMessageHandler = new HttpClientHandler {MaxConnectionsPerServer = MinConnectionLimit };
             if (httpMessageHandler.SupportsAutomaticDecompression)
             {
                 httpMessageHandler.AutomaticDecompression =
                     useCompression ?
                         DecompressionMethods.GZip | DecompressionMethods.Deflate
                         : DecompressionMethods.None;
             }
-            else if (httpMessageHandler.SupportsAutomaticDecompression == false &&
```
Comment (ppekrol) [inline src/Raven.Client/Http/RequestExecutor.cs:1791]: why `httpMessageHandler.SupportsAutomaticDecompression == false` was removed?

Comment (haludi) [reply src/Raven.Client/Http/RequestExecutor.cs:1791]: Because the `else` handle it already, no? Or I miss here something...

Comment (ppekrol) [reply src/Raven.Client/Http/RequestExecutor.cs:1791]: Yes, you are correct. Didn't notice. My fault.

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 3]
```diff
@@ -38,6 +38,8 @@ namespace Raven.Client.Http
 {
     public class RequestExecutor : IDisposable
     {
+        private const int DefaultConnectionLimit = 1024 * 10;
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecutor.cs]: Can we make this public static value, so users have access to it?

**Pull request merged**